lib/class-wp-theme-json.php

```diff
@@ -271,25 +250,45 @@ class WP_Theme_JSON {
 		),
 	);
 
+	const ELEMENTS = array(
+		'link' => 'a:not(.wp-block-button_link)',
```

Comment:
Having :not(.wp-block-button_link). increases the selector specificity and it seems strange that the global styles code is aware of the button block. Would it be possible to remove the not from here and have some styles on the button block to overwrite this style?

Reply:
Note that this is how it already worked for user styles (block-level) because it was part of [block-library/common.scss](https://github.com/WordPress/gutenberg/pull/30541/files#diff-96306a997c6ce5c61d0a2e5256ac17b8639613f598c02f5400a37a84f9ccd3ef). However, it did not work like that for the global stylesheet. Not sure why the difference but judging at the PR that introduced it seemed to be a legit use case to (covering against the button block) https://github.com/WordPress/gutenberg/pull/29557 cc @scruffian @aristath

Reply:
Removed the negation at https://github.com/WordPress/gutenberg/pull/31497